active-mq-artemis/src/main/scala/jms4s/activemq/activeMQ.scala

```diff
@@ -0,0 +1,54 @@
+package jms4s.activemq
+
+import cats.data.NonEmptyList
+import cats.effect.{ Blocker, Resource, Sync }
+import cats.implicits._
+import io.chrisdavenport.log4cats.Logger
+import javax.jms.Connection
+import jms4s.jms.JmsConnection
+import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory
+
+object activeMQ {
+
+  case class Config(
+    endpoints: NonEmptyList[Endpoint],
+    username: Option[Username] = None,
+    password: Option[Password] = None,
+    clientId: String
+  )
+  case class Username(value: String) extends AnyVal
+  case class Password(value: String) extends AnyVal
+  case class Endpoint(host: String, port: Int)
+
+  def makeConnection[F[_]: Sync: Logger](config: Config, blocker: Blocker): Resource[F, JmsConnection[F]] =
+    for {
+      connection <- Resource.make(
+                     Logger[F].info(s"Opening Connection to MQ at ${hosts(config.endpoints)}...") *>
+                       Sync[F].delay {
+                         val connectionFactory: ActiveMQConnectionFactory =
+                           new ActiveMQConnectionFactory(hosts(config.endpoints))
+                         connectionFactory.setClientID(config.clientId)
+
+                         val connection: Connection = config.username.map { (username) =>
+                           connectionFactory.createConnection(
+                             username.value,
+                             config.password.map(_.value).getOrElse("")
+                           )
```

Comment:
val connection: Connection = config.username.fold(connectionFactory.createConnection) {
                           (username) =>
                             connectionFactory.createConnection(
                               username.value,
                               config.password.map(_.value).getOrElse("")
                             )
                         }

I'd prefer something like this...but it's up to you